manual/manual/refman/lex.etex

```diff
@@ -1,6 +1,13 @@
 \section{Lexical conventions}
 \pdfsection{Lexical conventions}
 %HEVEA\cutname{lex.html}
+\subsubsection*{Character set}
+
+The character set of OCaml source code files is Unicode, encoded as
+UTF-8. However, codepoints outside of the ASCII character set are
+only permitted inside comments and string literals. (Non-ASCII
```

Comment:
Here it should be added something like "however the compiler does not check UTF-8 validity of the sources". This is important so that people do not start relying on invariants that do not exist (e.g. my string literals are UTF-8 valid because the compiler parsed them).

Reply:
@dbuenzli I can also add code to the lexer to actually check that it is valid UTF-8. This has been asked for several times above, so maybe that makes sense. If it doesn't get added, I agree it should be documented.